### DISCUSSION TITLE
Reticulum over Esp32 wifi esp-now mesh using zh_network (unlimited nodes)

### DISCUSSION THREAD

**Discussion by sloev** [markqvist/Reticulum#524] [Ideas]
Hi
I would suggest to use the project:

https://github.com/aZholtikov/zh_network

When thinking about creating esp32 esp-now based topologies using wifi as medium.

Atleast we could emulate the rnode but just propagate using modded (zh network) esp now transport and connect to phone or pc using bluetooth or wifi sta.

Zh network features FYI @markqvist 

The maximum size of transmitted data is up to 218 bytes.

Support of any data types.

All nodes are not visible to the network scanner.

Not required a pre-pairings for data transfer.

Broadcast or unicast data transmissions.

There are no periodic/synchronous messages on the network. All devices are in "silent mode" and do not "hum" into the air.

Each node has its own independent routing table, updated only as needed.

Each node will receive/send a message if it "sees" at least one device on the network.

The number of devices on the network and the area of use is not limited.

Possibility uses WiFi AP or STA modes at the same time with ESP-NOW.

**Comment by faragher** [comment]
> The maximum size of transmitted data is up to 218 bytes.

This is a blocker. Reticulum requires an MDU of 500 bytes.

You'd need to run at least three packets along with the logic for sequencing and reassembly for each Reticulum packet. I'm not saying it can't be done, but it's a bad fit, and the results would be poor. We had similar conceptual problems when people tried to use Meshtastic as a Reticulum bridge.

On the other hand, if you're simply looking to use LXMF messengers over the network, that could be done. You'd have to write your own routers, but the format is network stack agnostic.

**Comment by markqvist** [comment]
Using 2.4 GHz RF tech like ESP-Now has been on my wish-list for the RNode firmware for some time, but I haven't had time to implement it yet. It's more likely than not to become a reality sometime in the future, but I'll probably prefer writing my own PHY layer for it directly, instead of using a pre-built protocol, since it will be more efficient and can be more tailored to Reticulum that way.
